korim/src/commonMain/kotlin/com/soywiz/korim/bitmap/BitmapSlice.kt

```diff
@@ -5,11 +5,19 @@ import com.soywiz.kmem.*
 import com.soywiz.korio.resources.*
 import com.soywiz.korma.geom.*
 
+/**
+ * @property virtFrame This defines a virtual frame [RectangleInt] which surrounds the bounds [RectangleInt] of the [Bitmap].
+ *                     It is used e.g. in a trimmed texture atlas to specify the original size of a single texture.
+ *                     X and y of virtFrame is the offset of the virtual frame to the top left edge of
+ *                     the bounds rectangle. Width and height defines the size of the virtual frame.
+ */
 abstract class BmpSlice(
     val bmpBase: Bitmap,
     val bounds: RectangleInt,
     val name: String? = null,
-    val rotated: Boolean = false
+    val rotated: Boolean = false,
+    val trimmed: Boolean = false,
+    val virtFrame: RectangleInt = RectangleInt(0, 0, bounds.width, bounds.height)
```

Comment:
Maybe virtFrame could be nullable, then trimmed be something like:
```
val trimmed get() = virtFrame != null
```

Could we also precompute the virtual width/height in both cases (trimmed and not trimmed) and store it in this class to improve CPU-cache locality / avoid one indirection when using this info?